### DISCUSSION TITLE
New AI models VS Docker-Volumes and iSpy Endpoint config

### DISCUSSION THREAD

**Discussion by MrSiO** [doitandbedone/ispyagentdvr-docker#545] [Q&A]
Hi all.  Basic question.
I want to add a custom model to DeepStack and make sure iSpy is configured correctly to use it.

The endpoint in my iSpyDVR is   **/v1/vision/detection**
But I do not have that folder path on my deepstack docker host.
I have a mounted volume of:

   `volumes: - /home/user1/deepstack_data/Data/pytorch-models:/modelstore/detection`

**Q:** Is the /v1/vision/detection binded to /modelstore/detection ?

If so, I guess I should create a folder /modelstore/detection/custom/combined where I will put the combiend.pt file.
Then in iSpy, the endopoint would be **/v1/vision/????**

I'm kinda confused about the iSpy endpoint config VS my volume on container.... :(
Any help or example would be great.

Thx!

**Comment by ispysoftware** [comment]
I've got a major update to Agent's AI code coming very soon. Recommend you upgrade to using CodeProject's senseAI as deepstack seems to have been abandoned.
senseAI is based off deepstack and should support your custom models

I think the endpoint should be like
/v1/vision/custom/combined

but in the meantime:
https://www.ispyconnect.com/userguide-agent-deepstack-ai.aspx#customModels

https://www.codeproject.com/Articles/5348417/How-to-Use-a-Custom-Model-in-CodeProject-AI-Server

**Reply by Dvalin21** [reply]
If anyone hasn't told you today, we appreciate your work.

**Comment by MrSiO** [comment]
Great news!
I will do the switch and test it out!
Keep us posted!
